src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.cs

```diff
@@ -32,10 +37,15 @@ internal partial class SyntacticClassificationTaggerProvider : ITaggerProvider
         public SyntacticClassificationTaggerProvider(
             IForegroundNotificationService notificationService,
             ClassificationTypeMap typeMap,
-            [ImportMany] IEnumerable<Lazy<IAsynchronousOperationListener, FeatureMetadata>> asyncListeners)
+            [ImportMany] IEnumerable<Lazy<ILanguageService, LanguageServiceMetadata>> languageServices,
+            [ImportMany] IEnumerable<Lazy<IAsynchronousOperationListener, FeatureMetadata>> asyncListeners,
+            [ImportMany] IEnumerable<Lazy<ILanguageService, ContentTypeLanguageMetadata>> contentTypes
+            )
         {
             _notificationService = notificationService;
             _typeMap = typeMap;
+            _editorClassificationLanguageServices = languageServices.Where(s => s.Metadata.ServiceType == typeof(IEditorClassificationService).AssemblyQualifiedName);
```

Comment:
Oh wow...   is that really how you have to do this?  This seems pretty awful  :)